### DISCUSSION TITLE
PendingIntent problem in Scheduler.java.

### DISCUSSION THREAD

**Discussion by meteoguzhan** [strongswan/strongswan#1151] [Q&A] [answered]
Hi Team,

When I try to publish my Android application, I get the following error.
https://i.ibb.co/KGD2906/Screen-Shot-2022-07-18-at-23-18-11.png

I leave below the solution suggested by Google.
https://support.google.com/faqs/answer/10437428

I found the problem file.
https://github.com/strongswan/strongswan/blob/master/src/frontends/android/app/src/main/java/org/strongswan/android/logic/Scheduler.java

But I don't know how to solve it. Thank you very much in advance for your help.

**Comment by tobiasbrunner** [comment, marked as answer]
I've not seen this yet. But maybe it's only generated for new publications. According to the linked support document it's not a blocking issue, though ("Fixing this issue is recommended but not mandatory. The publication status of your app will be unaffected by the presence of this issue.").

I guess `FLAG_IMMUTABLE` could be set but I don't think this can be fully fixed there because according to the comment in `createIntent()`, setting a component/class on the base Intent doesn't work with dynamic broadcast receivers (i.e. if `registerReceiver()` is used). But maybe Google has fixed that issue (it's been two years since that code was written).

**Reply by meteoguzhan** [reply]
Thank you very much for your reply.

**Reply by tobiasbrunner** [reply]
I've pushed a commit that sets `FLAG_IMMUTABLE` to the _android-updates_ branch. That seems to be necessary when targeting Android 12 anyway (the system aborts the app otherwise). But if they require a component/class that probably won't make a difference in regards to Google Play.

**Reply by meteoguzhan** [reply]
Thank you very much. Would it be enough if I get the commit of the link below to pull this fix to my own project? Or is there a fix with another commit?
https://github.com/strongswan/strongswan/commit/d0dd63d0f5de4595816e4c8de51f222c0e841491

**Reply by tobiasbrunner** [reply]
Sure, the other changes are not related.

**Reply by meteoguzhan** [reply]
Thank you very much for your attention and for making this change. Google play no longer shows such a warning and everything is working properly.

**Reply by tobiasbrunner** [reply]
OK, good to know.
